Dockerfile.windows.x86_64

```diff
@@ -0,0 +1,32 @@
+# escape=`
+
+ARG WIN_VER="ltsc2016"
+
+FROM mcr.microsoft.com/windows/servercore:$WIN_VER
+
+ADD https://aka.ms/vs/16/release/vs_buildtools.exe C:\TEMP\vs_buildtools.exe
+ADD https://win.rustup.rs/x86_64 C:\TEMP\rustup-init.exe
+ADD https://chocolatey.org/install.ps1 C:\TEMP\choco-install.ps1
+
+# Let's be explicit about the shell that we're going to use.
+SHELL ["cmd", "/S", "/C"]
+
+# Install Build Tools. A 3010 error signals that requested operation is
+# successfull but changes will not be effective until the system is rebooted.
+RUN C:\TEMP\vs_buildtools.exe --quiet --wait --norestart --nocache `
```

Comment:
Does this command require tools to be installed on the host where you're building the container?

I am not very sure how containers work on Windows. Do you have a reference that could make this PR easier to review?

Reply:
Other than Docker for Windows, nothing is needed on the host side. This command will run in a container when building the image. We're installing the Visual Studio build tools, which we're fetching a few steps before that.

Nothing is Windows specific about this image (except for the shell and escape character), you can almost ignore the fact that it's supposed to run on Windows.

Since you've mentioned it, there are basically two types of Windows containers: process containers and hyper-v containers. The former feel like Linux containers, having similar levels of isolation and ability to limit resources, although obviously using OS specific primitives (no cgroups, seccomp, selinux, namespaces, apparmor - mostly using Windows specific equivalents though). The later provide an additional virtualization layer, which I usually compare with Kata Containers. You're no longer sharing the host kernel, which among other things means that you're no longer constrained to use images for the same Windows build version.

If anyone would like to try this out, I recommend taking a quick look at the docker docs. Also, keep in mind that we need actual Windows containers, so no Linux containers running in a Hyper-V VM, to which docker just acts as a frontend (I think those are the default container types).